### PR TITLE
Resolving Index Resetting for tz_localize

### LINKED ISSUE #43080: BUG: dt.tz_convert() on categorical[datetime] resets index
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

### Code Sample

```python
import itertools
import pandas as pd

ts_list = pd.date_range(start='2021-01-01T02:00:00', periods=3, freq='1D').to_list()
df = pd.DataFrame(data=list(itertools.islice(itertools.cycle(ts_list), 9)),
                  index=[2, 4, 5, 6, 7, 8, 11, 14, 15],
                  columns=['pure_datetime'])
df['categorical_datetime'] = df['pure_datetime'].astype(pd.CategoricalDtype())

"""
>>> df
         pure_datetime categorical_datetime
2  2021-01-01 02:00:00  2021-01-01 02:00:00
4  2021-01-02 02:00:00  2021-01-02 02:00:00
5  2021-01-03 02:00:00  2021-01-03 02:00:00
6  2021-01-01 02:00:00  2021-01-01 02:00:00
7  2021-01-02 02:00:00  2021-01-02 02:00:00
8  2021-01-03 02:00:00  2021-01-03 02:00:00
11 2021-01-01 02:00:00  2021-01-01 02:00:00
14 2021-01-02 02:00:00  2021-01-02 02:00:00
15 2021-01-03 02:00:00  2021-01-03 02:00:00

>>> df.dtypes
pure_datetime           datetime64[ns]
categorical_datetime          category
dtype: object
"""

# Now apply some function from .dt namespace.
"""
>>> df['pure_datetime'].dt.tz_localize('Europe/Berlin')
2    2021-01-01 02:00:00+01:00
4    2021-01-02 02:00:00+01:00
5    2021-01-03 02:00:00+01:00
6    2021-01-01 02:00:00+01:00
7    2021-01-02 02:00:00+01:00
8    2021-01-03 02:00:00+01:00
11   2021-01-01 02:00:00+01:00
14   2021-01-02 02:00:00+01:00
15   2021-01-03 02:00:00+01:00
Name: pure_datetime, dtype: datetime64[ns, Europe/Berlin]
"""
# All is fine with pure datetime column

# BUT, when applied to categorical column, the index is reset:
"""
>>> df['categorical_datetime'].dt.tz_localize('Europe/Berlin')
0   2021-01-01 02:00:00+01:00
1   2021-01-02 02:00:00+01:00
2   2021-01-03 02:00:00+01:00
3   2021-01-01 02:00:00+01:00
4   2021-01-02 02:00:00+01:00
5   2021-01-03 02:00:00+01:00
6   2021-01-01 02:00:00+01:00
7   2021-01-02 02:00:00+01:00
8   2021-01-03 02:00:00+01:00
Name: categorical_datetime, dtype: datetime64[ns, Europe/Berlin]
"""

# Naive construction of new columns yields convoluted results
df['localized_pure'] = df['pure_datetime'].dt.tz_localize('Europe/Berlin')
df['localized_categorical'] = df['categorical_datetime'].dt.tz_localize('Europe/Berlin')
"""
>>> df
         pure_datetime categorical_datetime            localized_pure     localized_categorical
2  2021-01-01 02:00:00  2021-01-01 02:00:00 2021-01-01 02:00:00+01:00 2021-01-03 02:00:00+01:00
4  2021-01-02 02:00:00  2021-01-02 02:00:00 2021-01-02 02:00:00+01:00 2021-01-02 02:00:00+01:00
5  2021-01-03 02:00:00  2021-01-03 02:00:00 2021-01-03 02:00:00+01:00 2021-01-03 02:00:00+01:00
6  2021-01-01 02:00:00  2021-01-01 02:00:00 2021-01-01 02:00:00+01:00 2021-01-01 02:00:00+01:00
7  2021-01-02 02:00:00  2021-01-02 02:00:00 2021-01-02 02:00:00+01:00 2021-01-02 02:00:00+01:00
8  2021-01-03 02:00:00  2021-01-03 02:00:00 2021-01-03 02:00:00+01:00 2021-01-03 02:00:00+01:00
11 2021-01-01 02:00:00  2021-01-01 02:00:00 2021-01-01 02:00:00+01:00                       NaT
14 2021-01-02 02:00:00  2021-01-02 02:00:00 2021-01-02 02:00:00+01:00                       NaT
15 2021-01-03 02:00:00  2021-01-03 02:00:00 2021-01-03 02:00:00+01:00                       NaT
"""

```

#### Problem description

When working on categorical columns, I would expect all methods, that can be called without error or warning, to yield the same results as when applied to the pure (category-expanded) columns. Some methods from the `.dt`-namespace (at least `tz_localize()` and `tz_convert()`) seem to return non-consistent results.
In particular, they reset the index of the input column (a Series in fact).

#### Expected Output

I expect the method `.dt.tz_localize()` to not reset the index when applied to a Series of datetime categoricals.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 5f648bf1706dd75a9ca0d29f26eadfbb595fe52b
python           : 3.9.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.14393
machine          : AMD64
processor        : Intel64 Family 6 Model 85 Stepping 0, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : de_DE.cp1252

pandas           : 1.3.2
numpy            : 1.21.2
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 21.2.4
setuptools       : 57.4.0
Cython           : None
pytest           : 6.2.4
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.1
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by kurchi1205**
- [x] closes #43080 
- [x] tests added / passed
   I have added a test function `test_series_tz_localize_matching_index`  in `test_tz_localize.py`
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry
   Index of the original series is preserved while converting `categorical` data to `datetime`

### CODE REVIEW

File: doc/source/whatsnew/v1.3.3.rst [hunk 1]
```diff
@@ -25,8 +25,8 @@ Fixed regressions
 
 Bug fixes
 ~~~~~~~~~
-- Bug in :meth:`.DataFrameGroupBy.agg` and :meth:`.DataFrameGroupBy.transform` with ``engine="numba"`` where ``index`` data was not being correctly passed into ``func`` (:issue:`43133`)
--
+- Bug in :meth:`.DataFrameGroupBy.agg` and :meth:`.DataFrameGroupBy.transform` with engine="numba" where index data was not being correctly passed into func (:issue:`43133`)
+- Fixed bug in :func:`__new__` of :class:`CombinedDatetimelikeProperties` maintains index of Categorical Datetime series (:issue:`43080`)
```
[the remark this answers is not in the document]
Comment (kurchi1205) [reply doc/source/whatsnew/v1.3.3.rst:29]: Ok I will make the necessary changes , 
Just wanted to be clear ,  why it shouldn't be in `v.1.3.3.rst` ?

[the remark this answers is not in the document]
Comment (mroeschke) [reply doc/source/whatsnew/v1.3.3.rst]: 1.3.3 is typically for regressions or bugs for very recently added features

File: doc/source/whatsnew/v1.4.0.rst [hunk 2]
```diff
@@ -310,7 +310,7 @@ Indexing
 - Bug in :meth:`Index.get_indexer_non_unique` when index contains multiple ``np.nan`` (:issue:`35392`)
 - Bug in :meth:`DataFrame.query` did not handle the degree sign in a backticked column name, such as \`Temp(°C)\`, used in an expression to query a dataframe (:issue:`42826`)
 - Bug in :meth:`DataFrame.drop` where the error message did not show missing labels with commas when raising ``KeyError`` (:issue:`42881`)
--
+- Bug in :meth:`dt.tz_convert` resetting index in a :class:`Series` with :class:`CategoricalIndex` (:issue:`43080`)
```
Comment (phofl) [inline doc/source/whatsnew/v1.4.0.rst:313]: I think Timezones would be a better section?

File: pandas/tests/series/methods/test_tz_localize.py [hunk 3]
```diff
@@ -41,6 +42,23 @@ def test_series_tz_localize_ambiguous_bool(self):
...
         tm.assert_series_equal(result, expected1)
 
+    def test_series_tz_localize_matching_index(self):
+        # Matching the index of the result with that of the original series
+        # GH 43080
+        dt_series = Series(
+            date_range(start="2021-01-01T02:00:00", periods=5, freq="1D"),
+            index=[2, 6, 7, 8, 11],
+        )
+        cat_series = dt_series.astype(CategoricalDtype())
```
Comment (phofl) [inline pandas/tests/series/methods/test_tz_localize.py:52]: Can you construct without astype?

**Review by phofl — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.4.0.rst (hunk 2 above)
Comment (kurchi1205) [reply doc/source/whatsnew/v1.4.0.rst:313]: the error was in indexing so I updated it in the indexing section

Comment (phofl) [reply doc/source/whatsnew/v1.4.0.rst:313]: Ok

Comment (kurchi1205) [reply doc/source/whatsnew/v1.4.0.rst]: I have already sent it timezone section

File: pandas/tests/series/methods/test_tz_localize.py (hunk 3 above)
Comment (kurchi1205) [reply pandas/tests/series/methods/test_tz_localize.py]: yeah , I have done that .

File: doc/source/whatsnew/v1.4.0.rst [hunk 4]
```diff
@@ -276,7 +276,7 @@ Timedelta
 
 Timezones
 ^^^^^^^^^
--
+- Bug in :meth:`dt.tz_convert` resetting index in a :class:`Series` with :class:`CategoricalIndex` (:issue:`43080`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:279]: this won't render, try Series.dt.tz_convert

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
this looks good. comment & pls merge master and ping on green.

**Comment by kurchi1205** [comment]
I have made the change in the documentation .

**Comment by kurchi1205** [comment]
I assume some maintainer has to merge the pull request.

**Comment by kurchi1205** [comment]
How to solve this timeout error ?

File: pandas/tests/series/methods/test_tz_localize.py [hunk 5]
```diff
@@ -41,6 +41,23 @@ def test_series_tz_localize_ambiguous_bool(self):
         result = ser.dt.tz_localize("US/Central", ambiguous=[False])
         tm.assert_series_equal(result, expected1)
 
+    def test_series_tz_localize_matching_index(self):
```
Comment (jreback) [inline pandas/tests/series/methods/test_tz_localize.py:44]: can you also add a test for another accessor (e.g. .hours) as i suspect that was also borked ; you can do this in parameterization or in another test if easier

Comment (kurchi1205) [reply pandas/tests/series/methods/test_tz_localize.py:44]: ok working on it

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm @phofl merge when good

**Comment by kurchi1205** [comment]
Can someone help me understand this failing test ? How are my changes affecting this test case ?

**Review by phofl — APPROVED** [review]
Lgtm failure unrelated

**Comment by phofl** [comment]
Thx @kurchi1205

**Comment by kurchi1205** [comment]
ok thnx for the clarification

**Pull request merged**